spec.bs

```diff
@@ -350,148 +340,135 @@ Each {{URLPattern}} object has an associated <dfn for=URLPattern>hash component<
 </div>
 
 <div algorithm>
   The <dfn attribute for="URLPattern">port</dfn> getter steps are:
 
-  1. Return [=this=]'s [=URLPattern/port component=]'s [=component/pattern string=].
+  1. Return [=this=]'s [=URLPattern/associated URL pattern=]'s [=URL pattern/port component=]'s [=component/pattern string=].
 </div>
 
 <div algorithm>
   The <dfn attribute for="URLPattern">pathname</dfn> getter steps are:
 
-  1. Return [=this=]'s [=URLPattern/pathname component=]'s [=component/pattern string=].
+  1. Return [=this=]'s [=URLPattern/associated URL pattern=]'s [=URL pattern/pathname component=]'s [=component/pattern string=].
 </div>
 
 <div algorithm>
   The <dfn attribute for="URLPattern">search</dfn> getter steps are:
 
-  1. Return [=this=]'s [=URLPattern/search component=]'s [=component/pattern string=].
+  1. Return [=this=]'s [=URLPattern/associated URL pattern=]'s [=URL pattern/search component=]'s [=component/pattern string=].
 </div>
 
 <div algorithm>
   The <dfn attribute for="URLPattern">hash</dfn> getter steps are:
 
-  1. Return [=this=]'s [=URLPattern/hash component=]'s [=component/pattern string=].
+  1. Return [=this=]'s [=URLPattern/associated URL pattern=]'s [=URL pattern/hash component=]'s [=component/pattern string=].
 </div>
 
 <div algorithm>
   The <dfn attribute for="URLPattern">hasRegExpGroups</dfn> getter steps are:
 
-  1. If [=this=] [=URLPattern/has regexp groups=], then return true.
+  1. If [=this=]'s [=URLPattern/associated URL pattern=]'s [=URL pattern/has regexp groups=], then return true.
   1. Return false.
 </div>
 
 <div algorithm>
   The <dfn method for="URLPattern">test(|input|, |baseURL|)</dfn> method steps are:
 
-  1. Let |result| be the result of [=URLPattern/match=] given [=this=], |input|, and |baseURL| if given.
+  1. Let |result| be the result of [=URL pattern/match=] given [=this=]'s [=URLPattern/associated URL pattern=], |input|, and |baseURL| if given.
   1. If |result| is null, return false.
   1. Return true.
 </div>
 
 <div algorithm>
   The <dfn method for="URLPattern">exec(|input|, |baseURL|)</dfn> method steps are:
 
-  1. Return the result of [=URLPattern/match=] given [=this=], |input|, and |baseURL| if given.
+  1. Return the result of [=URL pattern/match=] given [=this=]'s [=URLPattern/associated URL pattern=], |input|, and |baseURL| if given.
 </div>
 
-<h3 id=urlpattern-internals>Internals</h3>
-
-A {{URLPattern}} is associated with multiple <dfn>component</dfn> [=structs=].
+<h3 id=url-pattern-struct>The URL pattern struct</h3>
 
-A [=component=] has an associated <dfn for=component>pattern string</dfn>, a [=pattern string/well formed=] [=/pattern string=], which must be set upon creation.
+A <dfn>URL pattern</dfn> is a [=struct=] with the following [=struct/items=]:
 
-A [=component=] has an associated <dfn for=component>regular expression</dfn>, a {{RegExp}}, which must be set upon creation.
+* A <dfn for="URL pattern">protocol component</dfn>, a [=component=]
```

Comment:
nit: these leading "A"s confused me, why not simply:

```suggestion
* <dfn for="URL pattern">protocol component</dfn>, a [=component=]
```

and so on (here and also the items of "component")